velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -384,6 +392,107 @@ TEST_F(Re2FunctionsTest, likePattern) {
   EXPECT_EQ(like("abc", "MEDIUM POLISHED%"), false);
 }
 
+TEST_F(Re2FunctionsTest, likePatternFixed) {
+  auto like = [&](std::optional<std::string> str,
+                  std::optional<std::string> pattern) {
+    return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+  };
+
+  EXPECT_EQ(like("abcde", "abcde"), true);
+  EXPECT_EQ(like("ABCDE", "ABCDE"), true);
+  EXPECT_EQ(like("abcde", "uvwxy"), false);
+  EXPECT_EQ(like("ABCDE", "abcde"), false);
+
+  std::string input;
+  std::string input2;
+  getLargeString(input, 1000);
+  getLargeString(input2, 1000);
+  EXPECT_EQ(like(input, input), true);
+  EXPECT_EQ(like(input2, input2), true);
+  EXPECT_EQ(like(input, input2), false);
+}
+
+TEST_F(Re2FunctionsTest, likePatternPrefix) {
+  auto like = [&](std::optional<std::string> str,
+                  std::optional<std::string> pattern) {
+    return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+  };
+
+  EXPECT_EQ(like("abcde", "abcd%"), true);
```

Comment:
Would you add a test for empty and single-character input string?

Would you also add a tests for patterns with more than one % at the end and patterns that use _?